### PR TITLE
lib/packagekit: Use RHSM1 service

### PR DESCRIPTION
**Pull request by mvollmer**
The com.redhat.SubscriptionManager service is old and not always
available.

### CODE REVIEW

**Comment by martinpitt** [comment]
This now [fails tests on RHEL](https://logs.cockpit-project.org/logs/pull-13938-20200421-170434-d3b231aa-rhel-8-2-distropkg/log.html), so the tests need some way of masking/temporarily disabling the D-Bus service (replacing the old moving away of /usr/libexec/rhsmd)

File: pkg/lib/packagekit.js [hunk 1]
```diff
@@ -329,23 +324,35 @@ export function watchRedHatSubscription(callback) {
...
-                var sm = cockpit.dbus("com.redhat.SubscriptionManager");
+                var sm = cockpit.dbus("com.redhat.RHSM1");
+
+                // check if this is an unregistered RHEL system; if
+                // subscription-manager is not installed, ignore
+
+                function check() {
+                    sm.call(
+                        "/com/redhat/RHSM1/Entitlement", "com.redhat.RHSM1.Entitlement", "GetStatus", ["", ""])
+                            .done(result => {
```
Comment (martinpitt) [inline pkg/lib/packagekit.js:335]: Can you please use standard `.then`/`.done` JS promise API?

**Comment by marusak** [comment]
Nice, thanks @mvollmer !
Can you please also enable skips in tests for https://github.com/cockpit-project/cockpit/issues/13935 ?

File: pkg/lib/packagekit.js (hunk 1 above)
Comment (mvollmer) [reply pkg/lib/packagekit.js]: Then! Err, Done! :-)

**Comment by mvollmer** [comment]
> Can you please also enable skips in tests for #13935 ?

I don't understand, I am afraid... Shouldn't all tests be fixed now?

**Comment by mvollmer** [comment]
> > Can you please also enable skips in tests for #13935 ?
> 
> I don't understand, I am afraid... Shouldn't all tests be fixed now?

Ahh, I get it! :-) We shouldn't skip rhel-8-3 anymore, roger.

File: test/verify/check-packagekit [hunk 2]
```diff
@@ -45,8 +45,8 @@ class NoSubManCase(PackageCase):
         # Disable Subscription Manager on RHEL for these tests; subscriptions are tested in a separate class
         # On other OSes (Fedora/CentOS) we expect sub-man to be disabled in yum, so it should not get in the way there
         if self.machine.image in ["rhel-8-2", "rhel-8-2-distropkg", "centos-8-stream"]:
```
Comment (martinpitt) [inline test/verify/check-packagekit:47]: Fairly sure that we now need to add rhel-8-3 here. Or rather, `.startswith("rhel-")` and same for centos?

Comment (martinpitt) [reply test/verify/check-packagekit:47]: [Yep, we do](https://logs.cockpit-project.org/logs/pull-13938-20200424-055917-3817512b-rhel-8-3/log.html)

Comment (mvollmer) [reply test/verify/check-packagekit:47]: Note that this got changed just recently in fe1b8c6277a285990e45b477b2ef654f84e37f95

Comment (mvollmer) [reply test/verify/check-packagekit]: I have reverted part of fe1b8c6

Comment (martinpitt) [reply test/verify/check-packagekit]: Ah yes, that was a temporary hack to work around rhsmd not existing any more in 8.3. Thanks!

**Pull request merged**